packages/sitecore-jss-nextjs/src/editing/editing-render-middleware.ts

```diff
@@ -160,6 +163,11 @@ export class EditingRenderMiddleware {
       // The following line will trick it into thinking we're SSR, thus avoiding any router.replace.
       html = html.replace(STATIC_PROPS_ID, SERVER_PROPS_ID);
 
+      if (editingData.layoutData.sitecore.context.renderingType === RenderingType.Component) {
+        // Handle component rendering. Extract component markup only
+        html = parse(html).getElementById(EDITING_COMPONENT_ID)?.innerHTML;
```

Comment:
Should we throw an error here if we don't find the `EDITING_COMPONENT_ID`?

Reply:
For the edge case, yes - agree